src/main/java/com/cloudbees/dockerpublish/DockerBuilder.java

```diff
@@ -247,15 +240,30 @@ private Result executeCmd(String cmd) throws IOException, InterruptedException {
             PrintStream stdout = listener.getLogger();
             PrintStream stderr = listener.getLogger();
 
-            boolean result = launcher.launch()
-                    .envs(build.getEnvironment(listener))
-                    .pwd(build.getWorkspace())
-                    .stdout(stdout)
-                    .stderr(stderr)
-                    .cmdAsSingleString(cmd)
-                    .start().join() == 0;
-            String stdoutStr = stdout.toString();
-            return new Result(result, stdoutStr);
+            KeyMaterial registryKey = null;
+            try {
+                // get Docker registry credentials
+                registryKey = registry.materialize(build);
```

Comment:
Declare in initializer outside the `try` block, and then the `finally` block need not check for null.